chia_py_streamable_macro/src/lib.rs

```diff
@@ -38,72 +39,30 @@ pub fn py_streamable_macro(input: TokenStream) -> TokenStream {
-                    }
-                }
-                _ => {
-                    panic!("PyStreamable requires named fields");
-                }
+            for f in named.iter() {
+                fnames.push(f.ident.as_ref().unwrap().clone());
+                ftypes.push(f.ty.clone());
             }
 
-            quote! {
+            py_protocol.extend( quote! {
                 #[pymethods]
                 impl #ident {
                     #[new]
                     fn new ( #(#fnames : #ftypes),* ) -> #ident {
                         #ident { #(#fnames),* }
                     }
 
-                    #[staticmethod]
-                    pub fn from_bytes(blob: &[u8]) -> PyResult<Self> {
-                        let mut input = std::io::Cursor::<&[u8]>::new(blob);
-                        Self::parse(&mut input).map_err(|e| <chia_error::Error as Into<PyErr>>::into(e))
-                    }
-
-                    // returns the type as well as the number of bytes read from the buffer
-                    #[staticmethod]
-                    pub fn parse_rust<'p>(blob: pyo3::buffer::PyBuffer<u8>) -> PyResult<(Self, u32)> {
-                        if !blob.is_c_contiguous() {
-                            panic!("parse_rust() must be called with a contiguous buffer");
-                        }
-                        let slice = unsafe {
-                            std::slice::from_raw_parts(blob.buf_ptr() as *const u8, blob.len_bytes())
-                        };
-                        let mut input = std::io::Cursor::<&[u8]>::new(slice);
-                        Self::parse(&mut input).map_err(|e| <chia_error::Error as Into<PyErr>>::into(e)).map(|v| (v, input.position() as u32))
-                    }
-
-                    pub fn get_hash<'p>(&self, py: Python<'p>) -> PyResult<&'p pyo3::types::PyBytes> {
-                        let mut ctx = <clvmr::sha2::Sha256 as clvmr::sha2::Digest>::new();
-                        Streamable::update_digest(self, &mut ctx);
-                        Ok(pyo3::types::PyBytes::new(py, clvmr::sha2::Digest::finalize(ctx).as_slice()))
-                    }
-                    pub fn to_bytes<'p>(&self, py: Python<'p>) -> PyResult<&'p pyo3::types::PyBytes> {
-                        let mut writer = Vec::<u8>::new();
-                        self.stream(&mut writer).map_err(|e| <chia_error::Error as Into<PyErr>>::into(e))?;
-                        Ok(pyo3::types::PyBytes::new(py, &writer))
-                    }
-
-                    pub fn __bytes__<'p>(&self, py: Python<'p>) -> PyResult<&'p pyo3::types::PyBytes> {
-                        self.to_bytes(py)
-                    }
-
-                    pub fn __deepcopy__<'p>(&self, memo: &pyo3::PyAny) -> PyResult<Self> {
-                        Ok(self.clone())
-                    }
-
-                    pub fn __copy__<'p>(&self) -> PyResult<Self> {
-                        Ok(self.clone())
-                    }
-
```

Comment:
this block was moved out and down, to be reused between the two cases of named and unnamed fields. (it's called `streamable`)